### PR TITLE
new optional define IMGUI_IMPL_OPENGL_VERSION

### PR DESCRIPTION
**Pull request by dpilawa**
- new optional define `IMGUI_IMPL_OPENGL_VERSION` used in `imgui_impl_opengl3.cpp`
- if undefined attempt to automatically set based on GLAD defines
- improved `IMGUI_IMPL_OPENGL_HAS_DRAW_WITH_BASE_VERTEX` define (fixing #2852)

### CODE REVIEW

**Comment by dpilawa** [comment]
Thanks for your feedback, @ocornut !

Regarding regression - the `IMGUI_IMPL_OPENGL_HAS_DRAW_WITH_BASE_VERTEX` would actually be unset when neither User nor GLAD (indirectly by `GL_VERSION_`) defined `IMGUI_IMPL_OPENGL_VERSION`. So this is not a bug per se, but would effectively disable the use of `glDrawElementsBaseVertex` for anyone who does not set the new optional define explicitly, or uses GLAD. So yes, that's regression.

Now that we know that GLEW and GL3W use the same `GL_VERSION_` defines the problem would not be that big, but there is no guarantee anyway on the existence of these in the future. And there are still custom loaders.

So bottom line is - I agree with your suggestion to simplify and still use runtime safety-net in `ImGui_ImplOpenGL3_Init()` and in `ImGui_ImplOpenGL3_RenderDrawData()`. Let me get back to you with updated PR.

**Comment by ocornut** [comment]
Hi @dpilawa was just merging that now at the same time you updated the changelog :)
Will be merged in a minute, thanks again!
